xdp-loader/xdp-loader.c

```diff
@@ -139,6 +157,12 @@ int do_load(const void *cfg, __unused const char *pin_root_path)
 			goto out;
 		}
 
+		if (opt->prio)
+			__xdp_program__set_run_prio(p, opt->prio);
```

Comment:
This should check the return value and error out if it fails (and just use the `xdp_program__set_run_prio()` variant).